src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -911,14 +946,15 @@ impl<'a, 'tcx> DocFolder for LinkCollector<'a, 'tcx> {
 
 impl LinkCollector<'_, '_> {
     fn resolve_link(
-        &self,
+        &mut self,
```

Comment:
```suggestion
        &self,
```

Reply:
I don't see how, since caching in #77700 needs `mut self`, unless I introduce interior mutability which seems unnecessary.